### PR TITLE
Reduced db interactions

### LINKED ISSUE #131: Tourbook import takes forever
If I import a tourbook (about 250 entries) the app is busy for minutes. At the end the import is successful but for the user experience is not very nice. I see two tasks:
* Analyze why it takes so long and try to improve it
* Implement a progress bar so that the user at least as a feedback from the app

### PR DESCRIPTION
**Pull request by paetz**
Resolves #131 

This contains:
- Speeding up tourbook import
- Speeding up database update & deletions

Resolved by:
- decorating expensive database interactions with beginTransaction()
- hiding database behind wrapper that provides more efficient database access methods

### CODE REVIEW

File: app/src/main/java/com/yacgroup/yacguide/RockActivity.kt [hunk 1]
```diff
@@ -87,15 +87,15 @@ class RockActivity : UpdatableTableActivity() {
...
+        val comments = _sector?.let { db.getSectorComments(it.id) } ?: emptyList()
         for (comment in comments) {
             val qualityId = comment.qualityId
             val text = comment.text
 
             layout.addView(WidgetUtils.createHorizontalLine(this, 5))
             if (SectorComment.QUALITY_MAP.containsKey(qualityId)) {
                 layout.addView(WidgetUtils.createCommonRowLayout(this,
-                        "Bedeutung:",
+                        getString(R.string.importance),
```
Comment (chrgernoe) [inline app/src/main/java/com/yacgroup/yacguide/RockActivity.kt:98]: Maybe relevance is a bit better.

File: app/src/main/java/com/yacgroup/yacguide/RouteActivity.kt [hunk 2]
```diff
@@ -44,15 +42,15 @@ class RouteActivity : TableActivity() {
...
         when {
-            rockStatus == Rock.statusProhibited -> findViewById<TextView>(R.id.infoTextView).text = "Achtung: Der Felsen ist komplett gesperrt."
-            rockStatus == Rock.statusTemporarilyProhibited -> findViewById<TextView>(R.id.infoTextView).text = "Achtung: Der Felsen ist zeitweise gesperrt."
-            rockStatus == Rock.statusPartlyProhibited -> findViewById<TextView>(R.id.infoTextView).text = "Achtung: Der Felsen ist teilweise gesperrt."
-            _rock?.type == Rock.typeUnofficial -> findViewById<TextView>(R.id.infoTextView).text = "Achtung: Der Felsen ist nicht anerkannt."
+            rockStatus == Rock.statusProhibited -> findViewById<TextView>(R.id.infoTextView).setText(R.string.rock_fully_prohibited)
+            rockStatus == Rock.statusTemporarilyProhibited -> findViewById<TextView>(R.id.infoTextView).setText(R.string.rock_temporarily_prohibited)
+            rockStatus == Rock.statusPartlyProhibited -> findViewById<TextView>(R.id.infoTextView).setText(R.string.rock_partly_prohibited)
+            _rock?.type == Rock.typeUnofficial -> findViewById<TextView>(R.id.infoTextView).setText(R.string.rock_inofficial )
```
Comment (chrgernoe) [inline app/src/main/java/com/yacgroup/yacguide/RouteActivity.kt:53]: There is a unnecessary whitespace before the last bracket.

File: app/src/main/java/com/yacgroup/yacguide/SectorActivity.kt [hunk 3]
```diff
@@ -57,14 +54,14 @@ class SectorActivity : UpdatableTableActivity() {
...
-        val comments = _region?.let { db.regionCommentDao().getAll(it.id) } ?: emptyList()
+        val comments = _region?.let { db.getRegionsComments(it.id) } ?: emptyList()
         for (comment in comments) {
             val qualityId = comment.qualityId
 
             layout.addView(WidgetUtils.createHorizontalLine(this, 5))
             if (RegionComment.QUALITY_MAP.containsKey(qualityId)) {
                 layout.addView(WidgetUtils.createCommonRowLayout(this,
-                        "Bedeutung:",
+                        getString(R.string.importance),
```
Comment (chrgernoe) [inline app/src/main/java/com/yacgroup/yacguide/SectorActivity.kt:64]: Same as above.

File: app/src/main/java/com/yacgroup/yacguide/database/RockDao.kt [hunk 4]
```diff
@@ -17,31 +17,43 @@
...
+    fun getAllInRegion(regionId: Int): List<Rock>
+
+    @Query("${Rock.SELECT_ALL} ${Sector.JOIN_ON} Rock.parentId ${Region.JOIN_ON} Sector.parentId ${Region.FOR_COUNTRY} :countryName")
+    fun getAllInCountry(countryName: String): List<Rock>
+
+    @Query("${Rock.SELECT_ALL} WHERE id = :id")
     fun getRock(id: Int): Rock?
 
-    @Query("SELECT Rock.* FROM Rock JOIN Route ON Rock.id = Route.parentId JOIN Ascend ON Route.id = Ascend.routeId WHERE Ascend.id = :ascendId")
+    @Query("${Rock.SELECT_ALL} JOIN Route ON Rock.id = Route.parentId JOIN Ascend ON Route.id = Ascend.routeId WHERE Ascend.id = :ascendId")
```
Comment (chrgernoe) [inline app/src/main/java/com/yacgroup/yacguide/database/RockDao.kt:39]: Why do you not use the `JOIN_ON` constants here?

File: app/src/main/java/com/yacgroup/yacguide/network/RockParser.kt [hunk 5]
```diff
@@ -103,26 +104,28 @@ class RockParser(
...
         // update already ascended rocks
-        for (rock in db.rockDao().getAll(_sectorId)) {
-            for (ascend in db.ascendDao().getAscendsForRock(rock.id)) {
-                rock.ascendsBitMask = (rock.ascendsBitMask or AscendStyle.bitMask(ascend.styleId))
+        val updatedRocks = _db.getRocks(_sectorId).map {
+            for (ascend in _db.getRockAscends(it.id)) {
+                it.ascendsBitMask = it.ascendsBitMask or AscendStyle.bitMask(ascend.styleId)
             }
-            db.rockDao().setAscendsBitMask(rock.ascendsBitMask, rock.id)
+            it
```
Comment (chrgernoe) [inline app/src/main/java/com/yacgroup/yacguide/network/RockParser.kt:119]: What is that?

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -51,14 +51,37 @@
...
+    <string name="notes">Bemerkungen</string>
     <string name="first_ascend_unknown">Erstbegeher unbekannt</string>
     <string name="date_unknown">Datum unbekannt</string>
+    <string name="route_quality">Wegqualität:</string>
+    <string name="grade">Schwierigkeit:</string>
+    <string name="protection">Absicherung:</string>
+    <string name="drying">Abtrocknung:</string>
+    <string name="type_of_climbing">Kletterei:</string>
+    <string name="importance">Bedeutung:</string>
+    <string name="nature">Charakter:</string>
```
Comment (chrgernoe) [inline app/src/main/res/values/strings.xml:80]: I would add the colons where the strings are used. It is easier to add the colons where necessary as remove them, if not necessary.

File: app/src/main/java/com/yacgroup/yacguide/database/RockDao.kt (hunk 4 above)
Comment (paetz) [reply app/src/main/java/com/yacgroup/yacguide/database/RockDao.kt:39]: Because the Route.JOIN_ON refers to joining upwards on Route.Id. Here we join downwards on Route.parentId

File: app/src/main/java/com/yacgroup/yacguide/network/RockParser.kt (hunk 5 above)
Comment (paetz) [reply app/src/main/java/com/yacgroup/yacguide/network/RockParser.kt:119]: Kotlin-specific handler of the current element in the loop

File: app/src/main/res/values/strings.xml (hunk 6 above)
Comment (paetz) [reply app/src/main/res/values/strings.xml]: True but this would make usage much uglier: `${getString(R.string.my_string)}:`
Since we only use those strings together with colons, I would keep it to have better readable code.
